server/src/utils/brute/removeChoiceFromDestiny.ts

```diff
@@ -0,0 +1,103 @@
+import { randomBetween } from '@labrute/core';
+import {
+  Brute,
+  BruteStat,
+  DestinyChoiceType,
+  PetName,
+  PrismaClient,
+  SkillName,
+  WeaponName,
+} from '@labrute/prisma';
+
+export const removeChoiceFromDestiny = async (
+  prisma: PrismaClient,
+  brute: Pick<Brute, 'id'>,
+  ascended?: {
+    weapon?: WeaponName,
+    skill?: SkillName,
+    pet?: PetName
+  },
+) => {
+  const { weapon, skill, pet } = ascended || {};
+  const isWeapon = weapon && Object.values(WeaponName).includes(weapon as WeaponName);
+  const isSkill = skill && Object.values(SkillName).includes(skill as SkillName);
+  const isPet = pet && Object.values(PetName).includes(pet as PetName);
+
+  if (!isWeapon && !isSkill && !isPet) {
+    throw new Error('Invalid destiny choice type');
+  }
+
+  if ((isWeapon && isSkill) || (isWeapon && isPet) || (isSkill && isPet)) {
+    throw new Error('Invalid destiny choice type');
+  }
+
+  const destinyChoice = weapon || skill || pet;
+
+  if (!destinyChoice) {
+    throw new Error('Invalid destiny choice');
+  }
+
+  let destinyChoiceType: DestinyChoiceType;
+  if (isWeapon) {
+    destinyChoiceType = DestinyChoiceType.weapon;
+  } else if (isSkill) {
+    destinyChoiceType = DestinyChoiceType.skill;
+  } else {
+    if (!isPet) {
+      throw new Error('Invalid destiny choice type');
+    }
+    destinyChoiceType = DestinyChoiceType.pet;
+  }
+
+  let originalDestinyChoiceType: 'originalWeapon' | 'originalSkill' | 'originalPet';
+  if (isWeapon) {
+    originalDestinyChoiceType = 'originalWeapon';
+  } else if (isSkill) {
+    originalDestinyChoiceType = 'originalSkill';
+  } else {
+    if (!isPet) {
+      throw new Error('Invalid destiny choice type');
+    }
+    originalDestinyChoiceType = 'originalPet';
+  }
```

Comment:
No need for 2 sets of if/else, you can set originalDestinyChoiceType in the first set of if/else